components/brave_wallet/renderer/js_cardano_wallet_api.h

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_RENDERER_JS_CARDANO_WALLET_API_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_RENDERER_JS_CARDANO_WALLET_API_H_
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "content/public/renderer/render_frame.h"
+#include "content/public/renderer/render_frame_observer.h"
+#include "gin/arguments.h"
+#include "gin/wrappable.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote.h"
+
+namespace brave_wallet {
+
+// https://cips.cardano.org/cip/CIP-30
```

Comment:
Let's also briefly what part of CIP is implemented by this class. Same for js_cardano_provider.h